src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/ReclusterServiceIntTest.kt

```diff
@@ -30,15 +34,93 @@ class ReclusterServiceIntTest : IntegrationTestBase() {
       personKeyEntity = personKeyEntity,
     )
 
-    val cluster = await untilNotNull { personKeyRepository.findByPersonId(personKeyEntity.personId) }
-    reclusterService.recluster(cluster)
+    reclusterService.recluster(personKeyEntity.personId)
 
+    checkTelemetry(
+      TelemetryEventType.CPR_RECLUSTER_STARTED,
+      mapOf("UUID" to personKeyEntity.personId.toString()),
+    )
     checkTelemetry(
       TelemetryEventType.CPR_RECLUSTER_UUID_MARKED_NEEDS_ATTENTION,
       mapOf("UUID" to personKeyEntity.personId.toString()),
     )
   }
 
+  @Test
+  fun `should not recluster when single record that does not match`() {
+    val cro = randomCro()
+    val cluster1 = createPersonKey()
+    createPerson(
+      Person(
+        references = listOf(Reference(IdentifierType.CRO, cro)),
+        sourceSystemType = COMMON_PLATFORM,
+      ),
+      personKeyEntity = cluster1,
+    )
+    val matchResponse = MatchResponse(matchProbabilities = mutableMapOf("0" to 0.99999999))
+    stubMatchScore(matchResponse)
+
+    reclusterService.recluster(cluster1.personId)
```

Comment:
I know this already exists and is a well established pattern. Is there anything testing this all the way through from receiving a merge event (if that is what triggers all this code)? If not, it might help to surface problems like the one we saw in production yesterday when everything (apparently) fell over on receipt of an unmerge event

Reply:
I believe this is triggered only from an update, this is not triggered yet. As I was waiting to finish dev on it all then wire it in. So all telemetry can be checked